multimodal/src/autogluon/multimodal/data/infer_types.py

```diff
@@ -371,6 +453,9 @@ def infer_label_column_type_by_problem_type(
     -------
     Column types with the label columns' types inferred from the problem type.
     """
+    if label_columns is None:
```

Comment:
Shall we check here if this is for matcher then return column_types otherwise raise an error?

Reply:
If `label_columns` is None in supervised training, it would throw errors in other places, e.g., computing the loss. If we add a condition to check if a matching problem, then we would need check more problem types in future to determine whether to throw errors, which is kind of a burden. I think no condition here is more general.